pkg/model/requests_processing.go

```diff
@@ -56,21 +58,49 @@ func (rps *requestsProcessingStock) EntitiesInStock() []*simulator.Entity {
 }
 
 func (rps *requestsProcessingStock) Remove() simulator.Entity {
-	return rps.delegate.Remove()
+	request := rps.delegate.Remove().(*requestEntity)
+	*rps.occupiedCPUCapacityMillis -= *request.utilizationForRequest
+	return request
 }
 
 func (rps *requestsProcessingStock) Add(entity simulator.Entity) error {
 	rps.numRequestsSinceLast++
+	request := *entity.(*requestEntity)
+
+	freeCPUCapacityMillis := *rps.totalCPUCapacityMillis - *rps.occupiedCPUCapacityMillis
+	if freeCPUCapacityMillis == 0 {
+		freeCPUCapacityMillis = 1
+	}
+	cpuTime := request.requestConfig.CPUTimeMillis * 1000 / freeCPUCapacityMillis
+	processingTime := cpuTime + request.requestConfig.IOTimeMillis
+	utilizationForRequest := cpuTime * freeCPUCapacityMillis / processingTime
+	*request.utilizationForRequest = utilizationForRequest
+	*rps.occupiedCPUCapacityMillis += utilizationForRequest
 
 	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
-	totalTime := calculateTime(rps.delegate.Count(), rps.replicaMaxRPSCapacity, time.Second, rng)
-
-	rps.env.AddToSchedule(simulator.NewMovement(
-		"complete_request",
-		rps.env.CurrentMovementTime().Add(totalTime),
-		rps,
-		rps.requestsComplete,
-	))
+
+	//calculate currentUtilization in percentage
+	currentUtilization := *rps.occupiedCPUCapacityMillis * 100 / *rps.totalCPUCapacityMillis
+	totalTime := calculateTime(currentUtilization, time.Duration(processingTime)*time.Millisecond, rng)
+
+	if totalTime > request.requestConfig.Timeout {
+		*rps.occupiedCPUCapacityMillis -= utilizationForRequest
```

Comment:
Yes, you're right. We decided to ignore it at the beginning, but we're going to add the logic to include CPU usage for the duration of the timeout. I already suggested the algorithm (see http://go/oss-modeling-cpu-in-skenario).

As for the second question, subtraction happens in Remove() method RequestsProcessingStock

Reply:
Sounds good to approach this iteratively (in later changes). A comment or a TODO is nice in such situation, explaining the known issue. E.g. // TODO: This is inaccurate because, <explanation>. This can be fixed by <rough one-sentence plan, and/or a link>

For the second question, a comment might be helpful too, since it wasn't clear why subtraction in one case, but not the other.

As a general advice: if there are questions in reviews, it's often useful to answer it in the code itself in addition to (or instead of) review threads. When reviewers ask something, it's not so much about educating that one reviewer, but often more importantly it's a signal that something wasn't clear in the code. Of course, it's just a general approach -- no need to answer every single review question in the code, e.g. if you feel that the code is already clear enough, but just a rule of thumb to consider. :)